subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -1759,24 +1759,35 @@ int nrf_cloud_format_cell_pos_req_json(struct lte_lc_cells_info const *const inf
 
 	lte_array = cJSON_AddArrayToObjectCS(req_obj_out, NRF_CLOUD_CELL_POS_JSON_KEY_LTE);
 	if (!lte_array) {
+		err = -ENOMEM;
 		goto cleanup;
 	}
 
-	/* Add the current cell to the array */
-	lte_obj = add_lte_inf(lte_array, &inf->current_cell);
-	if (!lte_obj) {
-		goto cleanup;
-	}
+	/* Add the current cell to the array; if using a GCI search type, sometimes
+	 * there is no current cell.
+	 */
+	if (inf->current_cell.id != LTE_LC_CELL_EUTRAN_ID_INVALID) {
+		lte_obj = add_lte_inf(lte_array, &inf->current_cell);
+		if (!lte_obj) {
+			err = -ENOMEM;
+			goto cleanup;
+		}
 
-	/* Add neighbor cells if present */
-	err = add_ncells(lte_obj, inf->ncells_count, inf->neighbor_cells);
-	if ((err == -EINVAL) || (err == -ENOMEM)) {
-		goto cleanup;
+		/* Add neighbor cells if present */
+		err = add_ncells(lte_obj, inf->ncells_count, inf->neighbor_cells);
+		if ((err == -EINVAL) || (err == -ENOMEM)) {
+			goto cleanup;
+		}
 	}
 
 	/* Skip GCI cells if not present */
 	if (!inf->gci_cells_count || !inf->gci_cells) {
-		return 0;
+		if (inf->current_cell.id == LTE_LC_CELL_EUTRAN_ID_INVALID) {
+			err = -ENODATA;
+		} else {
+			err = 0;
+		}
+		goto cleanup;
```

Comment:
do you still want to cleanup here even if `err = 0` was set?
going to cleanup will delete the stuff added to the cJSON obj.